### ISSUE TITLE
Support history.replace, in addition to history.push

### ISSUE THREAD

**Issue by cursedquail** [NLincoln/react-qparams#1]
I was thinking something like 

```js
<QueryParams>
	{query => <button onClick={() => query.replaceQuery({ a: 'b' })}>click</button>
</QueryParams>
```

**Comment by ChrisLincoln** [comment]
I wrote the code and created the PR before I read the above. Oops.  My interface isn't the same. #3

**Comment by cursedquail** [comment]
Closed with #4
